app/controlplane/api/controlplane/v1/group.proto

```diff
@@ -0,0 +1,157 @@
+
+// GroupService provides operations for managing groups within the system
+service GroupService {
+  // Create creates a new group with the specified name and description
+  rpc Create(GroupServiceCreateRequest) returns (GroupServiceCreateResponse) {}
+  // Get retrieves a specific group by its ID
+  rpc Get(GroupServiceGetRequest) returns (GroupServiceGetResponse) {}
+  // List retrieves a paginated list of groups, with optional filtering
+  rpc List(GroupServiceListRequest) returns (GroupServiceListResponse) {}
+  // Update modifies an existing group's attributes
+  rpc Update(GroupServiceUpdateRequest) returns (GroupServiceUpdateResponse) {}
+  // Delete removes a group from the system
+  rpc Delete(GroupServiceDeleteRequest) returns (GroupServiceDeleteResponse) {}
+  // ListMembers retrieves the members of a specific group
+  rpc ListMembers(GroupServiceListMembersRequest) returns (GroupServiceListMembersResponse) {}
+}
+
+// GroupServiceCreateRequest contains the information needed to create a new group
+message GroupServiceCreateRequest {
+  // Name of the group to create
+  string name = 1 [(buf.validate.field).string.min_len = 1];
+  // Description providing additional information about the group
+  optional string description = 2 [(buf.validate.field).ignore_empty = true];
+}
+
+// GroupServiceCreateResponse contains the newly created group
+message GroupServiceCreateResponse {
+  // The created group with all its attributes
+  Group group = 1;
+}
+
+// GroupServiceGetRequest contains the identifier for the group to retrieve
+message GroupServiceGetRequest {
+  // UUID of the group to retrieve
+  string group_id = 1 [(buf.validate.field).string.uuid = true];
+}
+
+// GroupServiceGetResponse contains the requested group information
+message GroupServiceGetResponse {
+  // The requested group with all its attributes
+  Group group = 1;
+}
+
+// GroupServiceListsRequest contains parameters for filtering and paginating group results
+message GroupServiceListRequest {
+  // Optional filter to search by group name
+  optional string name = 1;
+  // Optional filter to search by group description
+  optional string description = 2;
+  // Optional filter to search by member email address
+  optional string member_email = 3;
+  // Pagination parameters to limit and offset results
+  OffsetPaginationRequest pagination = 4;
+}
+
+// GroupServiceListsResponse contains a paginated list of groups
+message GroupServiceListResponse {
+  // List of groups matching the request criteria
+  repeated Group groups = 1;
+  // Pagination information for the response
+  OffsetPaginationResponse pagination = 2;
+}
+
+// GroupServiceUpdateRequest contains the fields that can be updated for a group
+message GroupServiceUpdateRequest {
+  // UUID of the group to update
+  string group_id = 1 [(buf.validate.field).string.uuid = true];
+  // New description for the group (if provided)
+  optional string description = 2 [(buf.validate.field).ignore_empty = true];
```

Comment:
why can't you update the name?

Reply:
To be honest, I had in mind the convention that we have with names and probably kicked in. I'll change it, there is no reason to not allow it.